index.js

```diff
@@ -24,45 +35,62 @@ module.exports = async function loader(source) {
     const contractsBuildDirectory = config.contracts_build_directory;
     const contractFileName = path.basename(contractFilePath);
     const contractName = params.contract || contractFileName.charAt(0).toUpperCase() + contractFileName.slice(1, contractFileName.length - 4);
-    const compiledContractPath = path
-      .resolve(contractsBuildDirectory, `${contractName}.json`);
+    const compiledContractPath = path.resolve(contractsBuildDirectory, `${contractName}.json`);
 
     // if loader is disabled do not compile/push/upgrade, but still serve .json contracts from file system.
     if (!disabled) {
-      // wait until compile/push/update is done
-      while (isZeppelinBusy) await wait(500);
+      // check if local version is installed
+      const isLocal = await packageExist(oz, cwd);
+      // check if global version is installed
+      const isGlobal = !!which.sync(oz, { nothrow: true });
+      // if not installed at all do nothing
+      if (isLocal || isGlobal) {
+        // wait until compile/push/update is done
+        while (isZeppelinBusy) await wait(500);
 
-      isZeppelinBusy = true;
+        isZeppelinBusy = true;
 
-      try {
-        const execOptions = {
-          cwd,
-          env: {
-            ...process.env,
-            // disable an interactive in ZeppelinOS by setting env variable to prevent blocking
-            ZOS_NON_INTERACTIVE: 'FULL',
-          },
-        };
-        // push new code into local blockchain
-        let result = await exec(`zos push --network ${network}`, execOptions);
-        // update a proxy contract
-        result = await exec(`zos update ${contractName} --network ${network}`, execOptions);
-      } finally {
-        // release the lock
-        isZeppelinBusy = false;
+        try {
+          const execOptions = {
+            cwd,
+            env: {
+              ...process.env,
+              // disable an interactive in ZeppelinOS by setting env variable to prevent blocking
+              ZOS_NON_INTERACTIVE: 'FULL',
+            },
+          };
+          // local has priority over global
+          const command = isLocal ? `npx ${oz}` : oz;
+          // push new code into local blockchain
+          let result = await exec(`${command} push --network ${network}`, execOptions);
+          // update a proxy contract
+          result = await exec(`${command} update ${contractName} --network ${network}`, execOptions);
+        } finally {
+          // release the lock
+          isZeppelinBusy = false;
+        }
+      } else {
+        notInstalledMessage = ` ${oz} is not installed either locally or globally.`;
```

Comment:
```suggestion
        notInstalledMessage = ` ${oz} is required to support solidity hot-loading. Please run "npm install ${oz}", or disable hot-loading.`;
```

Also, I'd just fail outright if `zos` is not found instead of serving the contract. If I followed the code correctly, if the contract being required was compiled at least once, then the user will never see this error message, but the contract will never be reloaded.

I think the best course of action is to just fail right away if we see `zos` is not installed, and suggest the user to either install `zos` or disable hot loading.

Reply:
Yeah. Actually makes sense. Especially considering that it is disabled by default right now.